docs/source/virtual_rainforest/core/data.md

```diff
@@ -0,0 +1,223 @@
+   loaded Data configuration - which is a set of named variables and source files - and
+   then just uses {meth}`~virtual_rainforest.core.data.Data.load_from_file` to try and
+   load each one.
+
+### Adding a data array directly
+
+Adding a  DataArray to a {class}`~virtual_rainforest.core.data.Data` method takes an
+existing DataArray object and then uses the built in validation to match the data onto
+core axes. So, for example, the grid used above has a spatial resolution and size:
+
+```{code-cell}
+grid
+```
+
+One of the validation routines for the core spatial axis takes a DataArray with `x` and
+`y` coordinates and checks that the data covers all the cells in a square grid:
+
+```{code-cell}
+temperature_data = DataArray(
+    np.random.normal(loc=20.0, size=(10, 10)),
+    name="temperature",
+    coords={"y": np.arange(5, 100, 10), "x": np.arange(5, 100, 10)},
+)
+
+temperature_data.plot();
+```
+
+That data array can then be added to the  loaded and validated:
+
+```{code-cell}
+data["temperature"] = temperature_data
+```
+
+The representation of the {class}`virtual_rainforest.core.data.Data` instance now shows
+the loaded variables:
+
+```{code-cell}
+data
+```
+
+A variable can be accessed from the `data` object using the variable name as a key, and
+the data is returned as an :class:`xarray.DataArray` object.
+
+Note that the `x` and `y` coordinates have been mapped onto the internal `cell_id`
+dimension used to label the different grid cells (see the [Grid](./grid.md)
+documentation for details).
+
+```{code-cell}
+# Get the temperature data
+loaded_temp = data["temperature"]
+
+print(loaded_temp)
+```
+
+You can check whether a particular variable has been validated on a given core axis
+using the {meth}`~virtual_rainforest.core.data.Data.on_core_axis` method:
+
+```{code-cell}
+data.on_core_axis("temperature", "spatial")
+```
+
+### Loading data from a file
+
+Data can be loaded directly from a file by providing a path to a supported file
+format and the name of a variable stored in the file. In this example below, the
+NetCDF file contains a variable `temp` with dimensions `x` and `y`, both of which
+are of length 10: it contains a 10 by 10 grid that maps onto the shape of the
+configured grid.
+
+```{code-cell}
+# Load data from a file
+file_path = Path("../../data/xy_dim.nc")
+data.load_from_file(file_path, file_var_name="temp")
+```
+
+```{code-cell}
+data
+```
+
+```{code-cell}
+data.on_core_axis("temp", "spatial")
+```
+
+### Loading data from a configuration
+
+The configuration files for a Virtual Rainforest simulation can include a data
+configuration section. This can be used to automatically load multiple datasets into
+a Data object. The configuration file is TOML formatted and should contain an entry
+like the example below for each variable to be loaded.
+
+```toml
+[[core.data.variable]]
+file="'../../data/xy_dim.nc'"
+file_var_name="temp"
+data_var_name="another_name"
```

Comment:
Mostly for traceability. If a user uses remotely-sensed data from a known source with well defined documentation and naming structure, then having that file unaltered helps make a clearly traceable mapping of data from source to input. @vgro - thoughts?

Reply:
I guess this goes back to the general question of how much we expect the user to preprocess the data before using the VR. I think we agreed that we want the user to provide data in the right format (including correct variable names) and outsource any preprocessing (reprojecting,downscaling/,..). This puts the responsibility for correct inputs on the user rather than us trying to anticipate all possible formats, sources, errors and so on that they could come up with and trying to correct them.

Reply:
Yeah - fair point. I had been primarily thinking about the manipulation of the data structure (repro/rescaling) but agree that - as long as we have clear documentation of the required variable names by model etc - then assuming variable names are correct is simpler.